.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,36 @@
+<!--- Provide a general summary of your changes in the Title above -->
+<!--- Please include JIRA number in the commit message of the PR. -->
+<!--- This will link your PR to the respective JIRA automatically. -->
+
+Many thanks for submitting your Pull Request ❤️!
+
+Please make sure that your PR meets the following requirements:
+
+- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md)
+- [ ] Pull Request title is properly formatted: `RHOAIENG-XYZ Subject`
+- [ ] Pull Request contains link to the JIRA issue
+- [ ] Pull Request contains link to any dependent or related Pull Request
+- [ ] Pull Request contains description of the issue
+
+## Description
+<!--- Describe your changes in detail -->
+
+## JIRA issue:
+<!--- Link your JIRA and related links here for reference. -->
+
+## How Has This Been Tested?
+<!--- Please describe in detail how you tested your changes. -->
+<!--- Include details of your testing environment, and the tests you ran to -->
+<!--- see how your change affects other areas of the code, etc. -->
+
+## Screenshot or short clip:
+<!--- Attach a screenshot or a short clip demonstrating the feature. -->
+
+## Merge criteria:
+<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+
+- [ ] Acceptance criteria of the JIRA are met.
+- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
```

Comment:
by default we have _Squash and merge_ enabled, i dont think we should make "the commits are squashed in a cohesive manner" as criteria. meaningful messages are good to keep

Reply:
"Commit message should describe what was change, why it was changed and the way how it was changed" (c) I do not remember document I read years ago :)

Reply:
@ykaliuta the "how" part is the commit itself... ;)

Reply:
> @ykaliuta the "how" part is the commit itself... ;)

:) well, it's a bit different when extented to high level implementation description and concerns taken into account